crates/win-api-wrappers/src/handle.rs

```diff
@@ -19,15 +19,44 @@ unsafe impl Send for Handle {}
 unsafe impl Sync for Handle {}
 
 impl Handle {
-    pub fn new(handle: HANDLE, owned: bool) -> Self {
+    // Wraps a Windows [`HANDLE`].
+    //
+    // # Safety
+    //
+    // When `owned` is `true`:
+    //
+    // - `handle` is a valid handle to an open object.
+    // - `handle` is not a pseudohandle.
+    // - The caller is actually responsible for freeing the `HANDLE` when the value goes out of scope.
+    //
+    // When `owned` is `false`: no outstanding precondition.
+    pub unsafe fn new(handle: HANDLE, owned: bool) -> Self {
         Self { raw: handle, owned }
     }
 
+    // Wraps an owned Windows [`HANDLE`].
+    //
+    // # Safety
+    //
+    // - `handle` is a valid handle to an open object.
+    // - `handle` is not a pseudohandle.
+    // - The caller is actually responsible for freeing the `HANDLE` when the value goes out of scope.
+    pub unsafe fn new_owned(handle: HANDLE) -> Self {
+        // SAFETY: Same preconditions as the called function.
+        unsafe { Self::new(handle, true) }
```

Comment:
The `Drop` implementation is assuming we constructed the `Handle` object in a sane way to call `CloseHandle`, but there is no way for us to verify that the handle is actually owned outside of the callsite. Conceptually, calling `Handle::new_owned(handle)` or `Handle::new(handle, true)` is like calling `CloseHandle` and inherits the safety preconditions.